### DISCUSSION TITLE
Domain verification is incorrectly generated on the wallet connect dashboard.

### DISCUSSION THREAD

**Discussion by judy9712** [WalletConnect/walletconnect-monorepo#4451] [Developer Support] [answered]
Hello, I created a new project on the WalletConnect dashboard. I set the homepage URL to https://dao.nnn.so/.  
However, the TXT record is generated for nnn.so, so domain verification is not successful.  
What should I do in this case...?
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/107378179/17182641-1f40-4cf3-937d-54af976ce2b6)

**Comment by glitch-txs** [comment]
this is correct it's taking "dao" as a subdomain which is supported.

**Reply by raviriley** [reply]
https://github.com/orgs/WalletConnect/discussions/4451#discussioncomment-9194107

**Reply by Cali93** [reply]
Hey @judy9712,

This is the expected behaviour, you need to set up the TXT record on your domain (nnn.so in this case), not on the subdomain.

**Comment by raviriley** [comment]
@glitch-txs this is actually **not correct** for some edge cases. For example, `.co.uk` is handled correctly but some other TLDs are not. One such edge case is the newly added `.it.com` TLD. WalletConnect Verify incorrectly looks for the DNS record at `it.com` even though `.it.com` is the TLD.

Also, there are cases where subdomains are where the app lies, so maybe it makes more sense overall for the DNS record to be located where the user specifies, not where WalletConnect Verify assumes it should be? That way when new TLDs come out they don't break the Verify API? Either way, `.it.com` needs to be patched.

**Reply by raviriley** [reply]
I created an issue for this @glitch-txs https://github.com/WalletConnect/walletconnect-monorepo/issues/4471

**Reply by glitch-txs** [reply]
I think you might have a point here, we'll investigate this and try to improve it. Thanks!

**Reply by raviriley** [reply]
Sweet! This is currently preventing anyone with a `.it.com` domain from using WalletConnect. Any chance a patch can be pushed ASAP similar to how `.co.uk` behaves?

**Reply by Cali93** [reply, marked as answer]
This has been fixed as mentioned here. https://github.com/WalletConnect/walletconnect-monorepo/issues/4471#issuecomment-2109951489
